File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/zig/F18.P2.zig
const std = @import("std");

fn probe() i32 {
// BEGIN PROBE F18.P2
const util = @import("util.zig");

const result = util.pubAdd(2, 3);
// END PROBE F18.P2
    return result;
}

pub fn main() !void {
    var threaded: std.Io.Threaded = .init_single_threaded;
    const io = threaded.io();
    var out_buf: [64]u8 = undefined;
    var out = std.Io.File.stdout().writer(io, &out_buf);
    try out.interface.print("{d}\n", .{probe()});
    try out.interface.flush();
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/rust/F18.P2.rs
fn probe() -> i32 {
// BEGIN PROBE F18.P2
mod util {
    pub fn pub_add(a: i32, b: i32) -> i32 {
        a + b + secret()
    }

    fn secret() -> i32 {
        1
    }
}

let result = util::pub_add(2, 3);
// END PROBE F18.P2
result
}

fn main() {
    println!("{}", probe());
}
